packages/core/src/enhance/impl.js

```diff
@@ -1,39 +1,59 @@
 'use strict'
 
 import R from 'ramda'
-import { memoize, time } from '../impl/util'
+import { memoize } from '../impl/util'
 import * as data from '../data'
 import * as zip from '../zip'
 
-export function enhancer(config) {
-  const { registry, elementZipper } = config
+export function extract(config) {
+  const { registry, elementZipper, minNumOfArgs, maxNumOfArgs } = config
 
-  const enhancersForKind = memoize(kind => {
-    const enhancers = registry.get(kind)
-    return enhancers.isEmpty() ? null : enhancers
-  })
+  const enhancersForKind = memoize(kind => registry.get(kind))
 
-  async function enhance(loc, context) {
-    const { elementZipper } = context
-
-    const kind = data.flow(loc, zip.value, data.kindOf)
-    const enhancers = enhancersForKind(kind)
-    if (enhancers != null) {
+  async function _extract(loc, context) {
```

Comment:
1. `extract` is a bad name for
2. the function does too much

Try to write several simpler functions, a couple partial applications of R.filter (`R.filter(f => f.length <= 1)` and `R.filter(f => f.length > 1); a couple to *run* an array of enhancers (with a read value and without) and one to *apply* the provided updates.

currently, the function is called extract, but it also runs the enhancers...